main/src/ca/uwaterloo/flix/language/ast/Symbol.scala

```diff
@@ -70,21 +70,21 @@ object Symbol {
     * Returns a fresh variable symbol for the given identifier.
     */
   def freshVarSym(ident: Name.Ident)(implicit flix: Flix): VarSym = {
-    new VarSym(flix.genSym.freshId(), ident.name, Type.freshVar(Kind.Star), Scopedness.Unscoped, ident.loc)
+    new VarSym(flix.genSym.freshId(), ident.name, UnkindedType.freshVar(loc = ident.loc), Scopedness.Unscoped, ident.loc)
```

Comment:
I wonder if its a problem that symbols carry unkinded types. This means that later we cannot rely on a symbol for its kind?

Reply:
After discussion- let us add a comement that var symbols are always of kind Star.